lib/puppet/provider/user/hpux.rb

```diff
@@ -19,14 +22,23 @@
     value !~ /\s/
   end
 
-  has_features :manages_homedir, :allows_duplicates, :manages_passwords
+  has_features :manages_homedir, :allows_duplicates, :manages_passwords, :manages_password_age, :manages_expiry
 
   def deletecmd
     super.insert(1,"-F")
   end
 
   def modifycmd(param,value)
-    super.insert(1,"-F")
+     cmd = super(param, value)
+     cmd << "-F"
+     if self.trusted == "Trusted"
+        cmd << ";"
```

Comment:
Thanks.  The self.trusted is the part I have not been able to mock out in a
spec file.
On Jan 14, 2015 7:04 PM, "Michael Smith" <notifications@github.com> wrote:

> In lib/puppet/provider/user/hpux.rb
> <https://github.com/puppetlabs/puppet/pull/3443#discussion-diff-22981701>:
>
> >
> >    def deletecmd
> >      super.insert(1,"-F")
> >    end
> >
> >    def modifycmd(param,value)
> > -    super.insert(1,"-F")
> > +     cmd = super(param, value)
> > +     cmd << "-F"
> > +     if self.trusted == "Trusted"
> > +        cmd << ";"
>
> I'll see about coming up with specs that work for this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3443/files#r22981701>.
>